packages/store-engine/src/main/engine/FileEngine.test.node.js

```diff
@@ -45,23 +45,56 @@ describe('FileEngine', () => {
       .then(done)
       .catch(done.fail));
 
-  describe('"resolvePath"', () => {
-    it('properly validates paths.', done => {
-      const PRIMARY_KEY = 'primary-key';
+  describe('"enforcePathRestrictions"', () => {
+    const windowsFolder = 'C:\\Users\\bart\\Documents\\Database\\';
+    const unixFolder = '/home/marge/test/';
 
-      Promise.all([
-        engine.resolvePath('../etc', PRIMARY_KEY).catch(error => error),
-        engine.resolvePath('..\\etc', PRIMARY_KEY).catch(error => error),
-        engine.resolvePath('.etc', PRIMARY_KEY).catch(error => error),
-        engine.resolvePath(TABLE_NAME, '../etc').catch(error => error),
-        engine.resolvePath(TABLE_NAME, '..\\etc').catch(error => error),
-        engine.resolvePath(TABLE_NAME, '.etc').catch(error => error),
-      ]).then(results => {
-        for (error of results) {
-          expect(error instanceof StoreEngineError.PathValidationError).toBe(true);
-        }
-        done();
-      });
+    it('allows dots inside of primary keys.', () => {
+      const tableName = 'amplify';
+      const primaryKey = 'z.storage.StorageKey.EVENT.LAST_DATE';
+      const actual = FileEngine.enforcePathRestrictions(path.join(unixFolder, tableName), primaryKey);
+      expect(actual).toBeUndefined();
+    });
+
+    it('allows empty strings.', () => {
+      const tableName = 'amplify';
+      const primaryKey = '';
+      const actual = FileEngine.enforcePathRestrictions(path.join(unixFolder, tableName), primaryKey);
+      expect(actual).toBeUndefined();
+    });
+
+    it('allows navigation within the restricted folder.', () => {
```

Comment:
Why should we allow navigation within the restriced folder? Do we have a use-case for that?

Reply:
We can now have tables and primary keys that contains dots but also slashes now.

Reply:
👍

Reply:
We already have the `it('allows dots inside of primary keys.')` test case. We should add another one for `it('allows slashes inside of primary keys.')` to highlight this use case. I think its a really important one. Thanks for bringing that up, Sabri!